cpu/nrf5x_common/include/periph_cpu.h

```diff
@@ -54,15 +54,18 @@ extern "C" {
 #define GPIO_PIN(x,y)       ((x & 0) | y)
 
 /**
- * @brief   Override GPIO pull register select values
+ * @brief   Override GPIO modes
  * @{
  */
-#define HAVE_GPIO_PP_T
+#define HAVE_GPIO_MODE_T
 typedef enum {
-    GPIO_NOPULL = 0,        /**< do not use internal pull resistors */
-    GPIO_PULLUP = 3,        /**< enable internal pull-up resistor */
-    GPIO_PULLDOWN = 1       /**< enable internal pull-down resistor */
-} gpio_pp_t;
+    GPIO_IN    = (0 << 0 | 0 << 1 | 0 << 2),    /**< IN */
+    GPIO_IN_PD = (0 << 0 | 0 << 1 | 1 << 2),    /**< IN with pull-down */
+    GPIO_IN_PU = (0 << 0 | 0 << 1 | 2 << 2),    /**< IN with pull-up */
```

Comment:
no, as bit 3+4 together describe the pull resistor configuration and the values 0, 1, 2 are the ones also used in the ref manual, I think this notation is quite straight forward!